src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs

```diff
@@ -342,6 +342,48 @@ public SslSettings(X509Certificate2 certificate, bool suppressValidation)
             SuppressValidation = suppressValidation;
         }
 
+        /// <summary>
+        /// Validates that the SSL certificate has an accessible private key.
+        /// Should be called before starting the server to ensure proper TLS configuration.
+        /// </summary>
+        /// <exception cref="ConfigurationException">
+        /// Thrown when certificate lacks private key or application cannot access it.
+        /// </exception>
+        public void ValidateCertificate()
+        {
+            if (Certificate == null)
+                return; // No SSL configured
+
+            if (!Certificate.HasPrivateKey)
+            {
+                throw new ConfigurationException(
+                    "SSL certificate does not have a private key. " +
+                    "Ensure certificate is installed with private key permissions.");
+            }
+
+            // Actually test private key access (not just presence)
+            try
+            {
+                using (var rsa = Certificate.GetRSAPrivateKey())
```

Comment:
We also need to check to see if `.GetECDsaPrivateKey()` returns null, need to monitor these in the future to make sure that these two private key types are the only ones that are supported by SSL.